doc/internal/man7/EVP_PKEY.pod

```diff
@@ -0,0 +1,55 @@
+=pod
+
+=head1 NAME
+
+EVP_PKEY - an internal description
+
+=head1 SYNOPSIS
+
+ #include "crypto/evp.h"
+
+ struct evp_pkey_st;
+
+=head1 DESCRIPTION
+
+I<This is not a complete description yet>
+
+B<EVP_PKEY> is a complex type that's essentially a container for
+private/public key key pairs, but has had other uses as well.
+
+=for comment "uses" could as well be "abuses"...
+
+It can contain the legacy form of keys -- i.e. pointers to the low
+level key types, such as B<RSA>, B<DSA> and B<EC> --, but also the
+provided form of keys -- i.e. pointers to provider side key data.
+Those two forms are mutually exclusive; an B<EVP_PKEY> instance can't
+contain both a key in legacy form and in provided form.  Regardless of
+form, this key is commonly refered to as the "origin".
+
+An B<EVP_PKEY> also contains a cache of provided copies of the key,
+each adapted for the provider that is going to use that copy to
+perform some operation.  This does of course only the implementation
+of the "origin" key supports exporting the key data, otherwise the
```

Comment:
This sentence is hard to understand..

This does of course only the implementation
of the "origin" key supports exporting the key data.

Reply:
Ah, my typical word drop (I seem to do that quite a bit...)

I've rephrased quite a bit, I hope it's become more understandable.